### PR TITLE
Add mmorph MFunctor instances

### PR DESCRIPTION
**Pull request by kmicklas**
I also cleaned up some `MonadTrans` instances that can be derived, since it's closely related to `MFunctor`.

### CODE REVIEW

File: src/Reflex/BehaviorWriter/Base.hs [hunk 1]
```diff
@@ -48,7 +49,18 @@ import Reflex.TriggerEvent.Class
 
 -- | A basic implementation of 'BehaviorWriter'.
 newtype BehaviorWriterT t w m a = BehaviorWriterT { unBehaviorWriterT :: StateT [Behavior t w] m a }
-  deriving (Functor, Applicative, Monad, MonadIO, MonadFix, MonadAsyncException, MonadException) -- The list is kept in reverse order
+  -- The list is kept in reverse order
```
[the remark this answers is not in the document]
Comment (3noch) [reply src/Reflex/BehaviorWriter/Base.hs:52]: I realize it was already there.

File: src/Reflex/Query/Base.hs [hunk 2]
```diff
@@ -291,8 +295,8 @@ withQueryT f a = do
   return result
 
 -- | Maps a function over a 'QueryT' that can change the underlying monad
-mapQueryT :: (forall b. m b -> n b) -> QueryT t q m a -> QueryT t q n a
-mapQueryT f (QueryT a) = QueryT $ mapStateT (mapEventWriterT (mapReaderT f)) a
+mapQueryT :: Monad m => (forall b. m b -> n b) -> QueryT t q m a -> QueryT t q n a
```
Comment (3noch) [inline src/Reflex/Query/Base.hs:298]: This *technically* makes the change breaking...

Comment (kmicklas) [reply src/Reflex/Query/Base.hs:298]: @3noch Do you think I should put it back? I can also make `hoist = mapQueryT`.

Comment (3noch) [reply src/Reflex/Query/Base.hs:298]: `hoist = mapQueryT' might be better. The next release will be breaking anyway, but if there's a way to minimize it we might as well. I'm also not actually worried about this particular thing breaking anyone, but I have to err on the side of pedantic.

Comment (kmicklas) [reply src/Reflex/Query/Base.hs]: Done.

**Comment by 3noch** [comment]
Thanks @kmicklas!

File: src/Reflex/BehaviorWriter/Base.hs (hunk 1 above)
[the remark this answers is not in the document]
Comment (3noch) [reply src/Reflex/BehaviorWriter/Base.hs:52]: This comment has to do with the `StateT [Behavior t w]`

**Pull request merged**